volatility3/framework/automagic/linux.py

```diff
@@ -59,65 +61,97 @@ def stack(
         for _, banner in layer.scan(
             context=context, scanner=mss, progress_callback=progress_callback
         ):
-            dtb = None
+            # No need to try stackers on the same banner more than once
+            if banner in seen_banners:
+                continue
+            else:
+                seen_banners.append(banner)
+
             vollog.debug(f"Identified banner: {repr(banner)}")
 
             isf_path = linux_banners.get(banner, None)
             if isf_path:
-                table_name = context.symbol_space.free_table_name("LintelStacker")
+                table_name = context.symbol_space.free_table_name("LinuxStacker")
                 table = linux.LinuxKernelIntermedSymbols(
                     context,
                     "temporary." + table_name,
                     name=table_name,
                     isf_url=isf_path,
                 )
                 context.symbol_space.append(table)
-                kaslr_shift, aslr_shift = cls.find_aslr(
-                    context, table_name, layer_name, progress_callback=progress_callback
-                )
-
-                layer_class: Type = intel.Intel
-                if "init_top_pgt" in table.symbols:
-                    layer_class = intel.Intel32e
-                    dtb_symbol_name = "init_top_pgt"
-                elif "init_level4_pgt" in table.symbols:
-                    layer_class = intel.Intel32e
-                    dtb_symbol_name = "init_level4_pgt"
-                else:
-                    dtb_symbol_name = "swapper_pg_dir"
-
-                dtb = cls.virtual_to_physical_address(
-                    table.get_symbol(dtb_symbol_name).address + kaslr_shift
-                )
-
-                # Build the new layer
-                new_layer_name = context.layers.free_layer_name("IntelLayer")
-                config_path = join("IntelHelper", new_layer_name)
-                context.config[join(config_path, "memory_layer")] = layer_name
-                context.config[join(config_path, "page_map_offset")] = dtb
+                new_layer_name = context.layers.free_layer_name("LinuxLayer")
+                config_path = path_join("LinuxHelper", new_layer_name)
+                context.config[path_join(config_path, "memory_layer")] = layer_name
                 context.config[
-                    join(config_path, LinuxSymbolFinder.banner_config_key)
+                    path_join(config_path, LinuxSymbolFinder.banner_config_key)
                 ] = str(banner, "latin-1")
 
-                layer = layer_class(
-                    context,
-                    config_path=config_path,
-                    name=new_layer_name,
-                    metadata={"os": "Linux"},
-                )
-                layer.config["kernel_virtual_offset"] = aslr_shift
+                linux_arch_stackers = [LinuxIntelSubStacker, LinuxAArch64SubStacker]
+                for linux_arch_stacker in linux_arch_stackers:
```

Comment:
Here, the stacking is operated blindly. However, we could also rely on symbols to determine the architecture to stack on. 

I identified `idmap_pg_dir` to be arm64 exclusive, and present since the first 3.7 release. Relying on a symbol, avoids "forced exception rounds", which shouts about missing symbols when an AArch64 image is first passed to the Intel stacker.